MTV-1433 UI plans by provider (#585)
kubev2v/forklift-documentation: documentation/modules/creating-migration-plan-2-8.adoc

====

include::snip_plan-limits.adoc[]

.Procedure

. In the {ocp} web console, click *Plans for virtualization* and then click *Create Plan*.
+
The *Create migration plan* wizard opens to the *Select source provider* interface.
. Select the source provider of the VMs you want to migrate.
+
The *Select virtual machines* interface opens.
. Select the VMs you want to migrate and click *Next*.
+
The *Create migration plan* pane opens. It displays the source provider's name and suggestions for a target provider and namespace, a network map, and a storage map.
. Enter the *Plan name*.
. To change the *Target provider*, the *Target namespace*, or elements of the *Network map* or the *Storage map*, select an item from the relevant list.
> To add either a *Network map* or a *Storage map*, click the *+* sign anf add a mapping.
. Click *Create migration plan*.
+
{project-short} validates the migration plan and the *Plan details* page opens, indicating whether the plan is ready for use or contains an error.
+
The details of the plan are listed, and you can edit the items you filled in on the previous page. If you make any changes, {project-short} validates the plan again.

ifdef::vmware[]

> . Check the following items related to VMware. All are optional except for *Warm migration*.

Reviewer: Following comments are relevant for all providers:
1. "All are optional except for.." - WDYM by a non optional settings? It's a bit confusing since all options, including this setting, have a default value. So if not set by the user, a default `cold` value is used. A non optional setting is usually something that the user have to set for proceeding and include a red small asterisk next to it, e.g. plan name

2. "`Transfer Network`" and "`Target namepsace`" settings are not detailed here. Is it in purpose? 
![Screenshot from 2025-01-20 14-30-44](https://github.com/user-attachments/assets/8e62d002-6629-4e9c-b84f-196faf59f192)

3. Creating a plan via the plan wizard can be done also by going to Provider -> VMs tab, selecting VM(s) => and clicking the "Create migration plan" button. Is it mentioned anywhere?

. Select the source provider of the VMs you want to migrate.
+
The *Select virtual machines* interface opens.
. Select the VMs you want to migrate and click *Next*.
+
The *Create migration plan* pane opens. It displays the source provider's name and suggestions for a target provider and namespace, a network map, and a storage map.
. Enter the *Plan name*.
. To change the *Target provider*, the *Target namespace*, or elements of the *Network map* or the *Storage map*, select an item from the relevant list.
. To add either a *Network map* or a *Storage map*, click the *+* sign anf add a mapping.
. Click *Create migration plan*.
+
{project-short} validates the migration plan, and the *Plan details* page opens, indicating whether the plan is ready for use or contains an error.
+
The details of the plan are listed, and you can edit the items you filled in on the previous page. If you make any changes, {project-short} validates the plan again.
+
. Check the following items in the *Settings* section of the page:

ifdef::vmware[]

* *Warm migration*: By default, all migrations are cold migrations. For a warm migration, click the *Edit* icon and select *Warm migration*.

> * *Transfer Network*: The network used to transfer the VMs to {virt}, by default, this is the default transfer network of the provider. To edit the transfer network, click the *Edit* icon, choose a different transfer network from the list in the window that opens, and click *Save*. 

Reviewer: One thing to add for the ***Transfer Network*** definition:
The chosen transfer network should be  in the selected ***target namespace***.

> * *Target namespace*: Destination namespace, by default, this is the {virt} cluster you chose as your destination provider. To edit the namespace, click the *Edit* icon, choose a different target namespace from the list in the window that opens, and click *Save*.  

Reviewer: Regarding **Target namespace** setting:
This is the destination namespace to be used by all the migrated VMs.
The default value for this item is the current/active namespace.

WDYM by "this is the OpenShift Virtualization cluster you chose as your destination provider"?

// * documentation/doc-Migration_Toolkit_for_Virtualization/master.adoc

:_mod-docs-content-type: PROCEDURE
[id="creating-migration-plan-2-8_{context}"]
= Creating a migration plan

Use the {ocp} web console to create a migration plan. Specify the source provider, the virtual machines (VMs) you want to migrate, and other plan details.

[WARNING]
====

> Do not include virtual machines with guest initiated storage connections, such as Internet Small Computer Systems Interface (iSCSI) connections or Network File System (NFS) mounts. These require either additional planning before a migration or reconfiguration after the migration

Reviewer 2: ```suggestion
Do not include virtual machines with guest-initiated storage connections, such as Internet Small Computer Systems Interface (iSCSI) connections or Network File System (NFS) mounts. These require either additional planning before migration or reconfiguration after the migration.
```

This prevents concurrent disk access to the storage the guest points to.
====

include::snip_plan-limits.adoc[]

.Procedure

. In the {ocp} web console, click *Plans for virtualization* and then click *Create Plan*.
+
The *Create migration plan* wizard opens to the *Select source provider* interface.
. Select the source provider of the VMs you want to migrate.
+
The *Select virtual machines* interface opens.
. Select the VMs you want to migrate and click *Next*.
+
The *Create migration plan* pane opens. It displays the source provider's name and suggestions for a target provider and namespace, a network map, and a storage map.
. Enter the *Plan name*.
. To change the *Target provider*, the *Target namespace*, or elements of the *Network map* or the *Storage map*, select an item from the relevant list.
> To add either a *Network map* or a *Storage map*, click the *+* sign anf add a mapping.
. Click *Create migration plan*.
+

> {project-short} validates the migration plan and the *Plan details* page opens, indicating whether the plan is ready for use or contains an error.

Reviewer 2: ```suggestion
{project-short} validates the migration plan, and the *Plan details* page opens, indicating whether the plan is ready for use or contains an error.
```
Conjunction joining two independent clauses means you need to use a comma

+
The details of the plan are listed, and you can edit the items you filled in on the previous page. If you make any changes, {project-short} validates the plan again.
+
. Check the following items in the *Settings* section of the page:

ifdef::vmware[]

> * *Warm migration*: By default, all migrations are cold migrations. For a warm migration, click the *Edit* icon, and select *Warm migration*.

Reviewer 2: ```suggestion
* *Warm migration*: By default, all migrations are cold migrations. For a warm migration, click the *Edit* icon and select *Warm migration*.
```
Not two independent clauses, so you do not need to use a comma

* *Transfer Network*: The network used to transfer the VMs to {virt}. By default... To edit the transfer network, click the *Edit* icon and...

> * *Target namespace*: Destination namespace, By default, this is the {virt} cluster you chose as your destination provider. To edit the namespace, click the *Edit* icon and...

Reviewer 2: ```suggestion
* *Target namespace*: Destination namespace, by default, this is the {virt} cluster you chose as your destination provider. To edit the namespace, click the *Edit* icon and...
```
Same as previous comment and the ellipses

+
{project-short} then issues a warning message about any VMs for which vNIC properties are missing. To retrieve any missing vNIC properties, run those VMs in vSphere in order for the vNIC properties to be reported to {project-short}.

* *Entering a list of decryption passphrases for disks encrypted using Linux Unified Key Setup (LUKS)*: To enter a list of decryption passphrases for LUKS-encrypted devices, in the *Settings* section, click the *Edit* icon next to *Disk decryption passphrases*, enter the passphrases, and then click *Save*. You do not need to enter the passphrases in a specific order. For each LUKS-encrypted device, {project-short} tries each passphrase until one unlocks the device.

* *Specifying a root device*: Applies to multi-boot VM migrations only. By default, {project-short} uses the first bootable device detected as the root device.
+

> To specify a different root device, in the *Settings* section, click the *Edit* icon next to *Root device* and choose a device from the list of commonly-used options, or enter a device in the text box.

Reviewer 2: it is not necessary in a compound modifier in which the first word is an adverb that ends in ‘ly’.
 
 https://www.merriam-webster.com/grammar/6-common-hypercorrections-and-how-to-avoid-them

+
{project-short} uses the following format for disk location: `/dev/sd<disk_identifier><disk_partition>`. For example, if the second disk is the root device and the operating system is on the disk's second partition, the format would be: `/dev/sdb2`. After you enter the boot device, click *Save*.
+
If the conversion fails because the boot device provided is incorrect, it is possible to get the correct information by checking the conversion pod logs.

endif::[]
ifdef::rhv[]

* *Warm migration* By default, all migrations are cold migrations. For a warm migration, click the *Edit* icon, and select *Warm migration*.

> * *Transfer Network*: The network used to transfer the VMs to {virt}. By default... To edit the transfer network, click the *Edit* icon and...

Reviewer 2: Clarify ellipses

> * *Target namespace*: Destination namespace, By default, this is the {virt} cluster you chose as your destination provider. To edit the namespace, click the *Edit* icon and...

Reviewer 2: ```suggestion
* *Target namespace*: Destination namespace, by default, this is the {virt} cluster you chose as your destination provider. To edit the namespace, click the *Edit* icon and...
```
Again, clarify the ellipses

* *Preserving the CPU model of VMs that are migrated from {rhv-short}*: Generally, the CPU model (type) for {rhv-short} VMs is set at the cluster level, but it can be set at the VM level, which is called a custom CPU model.
By default, {project-short} sets the CPU model on the destination cluster as follows:
** {project-short} preserves custom CPU settings for VMs that have them.
** For VMs without custom CPU settings, {project-short} does not set the CPU model. Instead, the CPU model is later set by {virt}.
+
To preserve the cluster-level CPU model of your {rhv-short} VMs, in the *Settings* section, click the *Edit* icon next to *Preserve CPU model*. Toggle the *Whether to preserve the CPU model* switch, and then click *Save*.
endif::[]

ifdef::ostack,ova,cnv[]

> * *Transfer Network*: The network used to transfer the VMs to {virt}. By default... To edit the transfer network, click the *Edit* icon and...

Reviewer 2: Clarify ellipses

> * *Target namespace*: Destination namespace, By default, this is the {virt} cluster you chose as your destination provider. To edit the namespace, click the *Edit* icon and...

Reviewer 2: ```suggestion
* *Target namespace*: Destination namespace, by default, this is the {virt} cluster you chose as your destination provider. To edit the namespace, click the *Edit* icon and...
```
Clarify ellipses

// * documentation/doc-Migration_Toolkit_for_Virtualization/master.adoc

:_mod-docs-content-type: PROCEDURE
[id="creating-migration-plan-2-8_{context}"]
= Creating a migration plan

> You can create a migration plan by using the {ocp} web console to specify a source provider, the virtual machines (VMs) you want to migrate, and other plan details.

Reviewer 3: Use the {ocp} web console to create a migration plan. Specify the source provider, the virtual machines (VMs) you want to migrate, and other plan details.

(I'm assuming the other plan details are obvious and/or optional.

[WARNING]
====

> Virtual machines with guest initiated storage connections, such as Internet Small Computer Systems Interface (iSCSI) connections or Network File System (NFS) mounts, are not handled by {project-short} and could require additional planning before or reconfiguration after the migration.

Reviewer 3: I’m not sure how to parse “not handled.”  Does this mean this type of connection fails, or perhaps {project-short} does not handle them well? The line that follows (“This is to ensure…) seems to indicate that they work but cause problems.

Since this is a warning, I’d say something like this:

Do not include virtual machines with guest initiated storage connections, such as Internet Small Computer Systems Interface (iSCSI) connections or Network File System (NFS) mounts. These  require either additional planning before a migration or reconfiguration after the migration.

Author: @fabiand Can you help here? Is Nunzy's suggested language OK?

Reviewer 4: +1 on Nunzy's wording.

> This is to ensure that no issues arise due to the addition or newly migrated VM accessing this storage.

Reviewer 3: It’s not clear to me what “due to the addition or newly migrated VM” means.  What’s the “addition VM”?  Does this mean this:

This ensures that no issues arise when a newly added VM or a newly migrated VM accesses this storage.

:-) Don't let me change the technical meaning if this is not what you are trying to say.

Author: @fabiand This change looks OK to me. WDYT?

Reviewer 4: Nunzy, thanks for finding this, but you do change the technical meaning, what about

> This is to ensure that no concurrent access to the storage arises after VM migration.

Author: @nunzy1 After chatting a bit, Fabian and I settled on "This prevents concurrent disk access to the storage the guest points to."

+
{project-short} validates the migration plan and the *Plan details* page opens, indicating whether the plan is ready for use or contains an error. The details of the plan are listed, and you can edit the items you filled in on the previous page. If you make any changes, {project-short} validates the plan again.

ifdef::vmware[]
+
[IMPORTANT]
====

> Check the following VMware-specific items. All are optional, but need special attention:

Reviewer 3: Check the following VMware items, all of which are optional:

(Do we need the second sentence, especially "special attention"?  This could be interpreted a variety of ways.)

ifdef::vmware[]

* *Warm migration*: By default, all migrations are cold migrations. For a warm migration, click the *Edit* icon and select *Warm migration*.
* *Transfer Network*: The network used to transfer the VMs to {virt}, by default, this is the default transfer network of the provider. Verify that the transfer network is in the selected target namespace.To edit the transfer network, click the *Edit* icon, choose a different transfer network from the list in the window that opens, and click *Save*. 

> * *Target namespace*: Ddestination namespace to be used by all the migrated VMs, by default, this is the current or active namespace. To edit the namespace, click the *Edit* icon, choose a different target namespace from the list in the window that opens, and click *Save*.  

Reviewer: Suggest to replace in all occurrences:

```suggestion
* *Target namespace*: Destination namespace to be used by all the migrated VMs, by default, this is the currently active namespace. To edit the namespace, click the *Edit* icon, choose a different target namespace from the list in the window that opens, and click *Save*.
```

+
If the conversion fails because the boot device provided is incorrect, it is possible to get the correct information by looking at the conversion pod logs.
====

endif::[]
ifdef::rhv[]
+
[IMPORTANT]
====
Check the following {rhv-short}-specific items. All are optional, but need special attention:

* *Preserving the CPU model of VMs that are migrated from {rhv-short}*: Generally, the CPU model (type) for {rhv-short} VMs is set at the cluster level, but it can be set at the VM level, which is called a custom CPU model.
By default, {project-short} sets the CPU model on the destination cluster as follows: {project-short} preserves custom CPU settings for VMs that have them, but, for VMs without custom CPU settings, {project-short} does not set the CPU model. Instead, the CPU model is later set by {virt}.
+
To preserve the cluster-level CPU model of your {rhv-short} VMs, in the *Settings* section, click the Edit icon next to *Preserve CPU model*. Toggle the *Whether to preserve the CPU model* switch, and then click *Save*.
====
endif::[]

. If your plan is valid:
.. You can run the plan now by clicking *Start migration*.

> .. You can run the plan later by selecting it on the *Plans for virtualization* page and following the procedure in xref:running-migration-plan_{context}[Running a migration plan].

Reviewer 3: Up to you...  I don't think you need "You can."

Author: @nunzy1 Here, I changed the lead-in.

endif::[]

ifdef::ostack,ova,cnv[]
* *Transfer Network*: The network used to transfer the VMs to {virt}, by default, this is the default transfer network of the provider. Verify that the transfer network is in the selected target namespace.To edit the transfer network, click the *Edit* icon, choose a different transfer network from the list in the window that opens, and click *Save*. 

> * *Target namespace*: Ddestination namespace to be used by all the migrated VMs, by default, this is the current or active namespace. To edit the namespace, click the *Edit* icon, choose a different target namespace from the list in the window that opens, and click *Save*.  

Reviewer: ```suggestion
this is the currently active namespace. 
```
